### ISSUE TITLE
💅 `domains.qwik: "none"` silently disables `correctness/useImageSize`

### ISSUE THREAD

**Issue by AlbinoGeek** [biomejs/biome#11879]
### Environment information

<details>

```bash
CLI:
  Version:                      2.5.14

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_DISTRIBUTION:           npm
  JS_RUNTIME_NAME:              bun
```

</details>

### Rule name

`correctness/useImageSize`

### Playground link

https://github.com/AlbinoGeek/biome-qwik-domain-suppresses-useimagesize

```tsx
export function Card({ src }: { src: string }) {
  return <img src={src} alt="cover" />;
}
```

```json
{ "linter": { "rules": { "preset": "all" }, "domains": { "qwik": "none" } } }
```

Setting `linter.domains.qwik` to `"none"` silently disables `correctness/useImageSize`,
a React and Next.js rule with no relationship to Qwik. The project declares `react` and
`next` as dependencies, so both domains are auto-detected.

| `linter.domains` | `useImageSize` | `noImgElement` |
| --- | --- | --- |
| `{}` | reports | reports |
| `{ "solid": "none" }` | reports | reports |
| `{ "qwik": "none" }` | **silent** | reports |
| `{ "next": "none" }` | reports | **silent** |

`next: "none"` suppressing `noImgElement` is correct. `qwik: "none"` suppressing
`useImageSize` is not, and no other domain value has this effect. Reproduced across
three consecutive runs with the daemon restarted between them.

Found because 36 repositories carried `domains: { qwik: "none", solid: "none" }` to
turn off framework rules they do not use, and none of them had been running
`useImageSize` as a result.

### Expected result

`useImageSize` reports, as it does with `domains: {}`. A domain value should only
affect rules belonging to that domain.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
I'm not sure I understand. That rule **belongs** to the qwick domain 

https://biomejs.dev/linter/domains/#qwik-rules

Where did you read about react and next?

**Comment by AlbinoGeek** [comment]
You're right, and I was wrong — sorry for the noise.

`biome explain useImageSize` lists exactly one domain:

```
Domains

- Name: qwik
- The rule is recommended for this domain
- The rule is enabled when one of these dependencies are detected:
  - @builder.io/qwik@>=1.0.0
  - @qwik.dev/core@>=2.0.0
```

So `qwik: "none"` turning it off is the setting doing its job. I read it as a React/Next rule because it governs `<img>` and our projects are React and Next — an assumption, not a measurement, and `explain` would have settled it in one command.

One thing I'd still like to understand before closing, since it's what led me astray. The rule fires in our projects at all only because of `preset: "all"`:

| config | `useImageSize` in a react+next project, no Qwik dependency |
| --- | --- |
| `recommended`, `domains: {}` | silent |
| `recommended`, `domains: { qwik: "none" }` | silent |
| `preset: "all"`, `domains: {}` | **reports** |
| `preset: "all"`, `domains: { qwik: "none" }` | silent |

Is that intended — that `preset: "all"` enables domain rules regardless of whether the domain's dependency is detected? It's defensible ("all means all"), but the effect is that a `preset: "all"` project runs framework rules for frameworks it doesn't use, and the only way to switch them back off is to name each domain `"none"` by hand. That's also what makes the domain setting look like it's reaching outside its own domain when it isn't.

Happy to close this as invalid either way — just say the word, or close it yourself if the answer is "yes, intended".
